workbench-core/example/infrastructure/src/example-hosting-stack.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-new */
+import {
+  WorkbenchEncryptionKeyWithRotation,
+  WorkbenchSecureS3Bucket
+} from '@aws/workbench-core-infrastructure';
+import { Aws, CfnResource, RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
+import { ArnPrincipal, PolicyStatement, Role } from 'aws-cdk-lib/aws-iam';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { CfnFunction } from 'aws-cdk-lib/aws-lambda';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { NagSuppressions } from 'cdk-nag';
+import { Construct } from 'constructs';
+import { createAccessLogsBucket } from './helpers/helper-function';
+
+export interface ExampleHostingStackProps extends StackProps {
+  mainAccountId: string;
+  lambdaRoleName: string;
+}
+
+export class ExampleHostingStack extends Stack {
+  public constructor(scope: Construct, id: string, props: ExampleHostingStackProps) {
+    super(scope, id, props);
+
+    const exampleHostingAccessLogsPrefix: string = 'example-hosting-access-log';
+
+    const accessLogsBucket: Bucket = createAccessLogsBucket(
+      this,
+      'ExampleHostingS3AccessLogsBucket',
+      exampleHostingAccessLogsPrefix,
+      'ExampleHostingAccessLogsNameOutput'
+    );
+
+    const exampleHostingEncryptionKey: WorkbenchEncryptionKeyWithRotation =
+      new WorkbenchEncryptionKeyWithRotation(this, 'ExampleHostingDataSetBucketEncryptionKey', {
+        removalPolicy: RemovalPolicy.DESTROY
+      });
+    const encryptionKey: Key = exampleHostingEncryptionKey.key;
+
+    const examplehostingDataSetBucket: Bucket = new WorkbenchSecureS3Bucket(
+      this,
+      'ExampleHostingDataSetBucket',
+      {
+        encryptionKey: encryptionKey,
+        serverAccessLogsBucket: accessLogsBucket,
+        serverAccessLogsPrefix: exampleHostingAccessLogsPrefix,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true
+      }
+    ).bucket;
+
+    const exampleCrossAccountRole = new Role(this, 'ExampleCrossAccountRole', {
```

Comment:
We need to use an `externalId` when using cross-account roles.

Reply:
Done